docs/js/examples.peggy

```diff
@@ -16,6 +16,8 @@ star = match:"a"* rest:rest { return {match, rest}; }
 
 plus = match:"a"+ rest:rest { return {match, rest}; }
 
+repetition = match:"a"|2..3, ","| rest:rest { return {match, rest}; }
```

Comment:
Playing around with this, it looks like this doesn't work as expected:

```peg.js
foo = n:num @"a" |n..n, "b" |n..n||

num = n:[0-9]+ { return parseInt(n, 10); }
```

Inputs: "2aba" matches, and I expected it to fail wanting another "b" before the last "a".  I assume this means that the inner range doesn't have visibility to the names from the outer block.  This isn't a big problem, but we should either fix it, find a work-around, or document it.

Reply:
This is a bug, looking for it...

Reply:
Seems to be fixed, but I want to check if more tests in parser behavior section is needed

Reply:
Added checks to `generate-bytecode.spec.js` and `generated-parser-behavior.spec.js`.